packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/PagespeedInsights.astro

```diff
@@ -0,0 +1,160 @@
+</div>
+
+<VisuallyHidden>
+    <div id="lighthouse-fetch-url">{FetchURL}</div>
+</VisuallyHidden>
+
+<style scoped>
+    #perf-ring, #acc-ring, #bp-ring, #seo-ring {
+        --track-width: 1rem;
+        font-weight: bolder;
+        font-size: 2rem;
+    }
+</style>
+
+<script>
+import type { SlButton, SlProgressRing } from "astrolace:types";
+
+function generateLighthouseFetchUrl(url: string, strategy: 'mobile' | 'desktop' | 'mixed-content' = 'mobile'): string {
+    // Base URL for Google PageSpeed Insights API
+    const baseUrl: string = 'https://www.googleapis.com/pagespeedonline/v5/runPagespeed';
+    
+    // Final URL with encoded parameters
+    const fetchUrl: string = `${baseUrl}?url=${encodeURIComponent(url)}&fields=lighthouseResult%2Fcategories%2F*%2Fscore&prettyPrint=false&strategy=${strategy}&category=performance&category=pwa&category=best-practices&category=accessibility&category=seo`;
+    
+    return fetchUrl;
+}
+
+const urlToTest: string = document.getElementById('lighthouse-fetch-url')?.textContent || '';
+
+const lighthouseFetchUrl: string = generateLighthouseFetchUrl(urlToTest);
+
+const runTestButton = document.getElementsByName('run-test')[0] as SlButton;
+
+runTestButton.addEventListener('click', async (e) => {
+    e.preventDefault();
+
+    runTestButton.loading = true;
+    runTestButton.disabled = true;
+
+    console.log('PageSpeed Insights test started.', lighthouseFetchUrl)
+
+    const response = await fetch(lighthouseFetchUrl);
+
+    let performanceScore = 0;
+    let accessibilityScore = 0;
+    let bestPracticesScore = 0;
+    let seoScore = 0;
+
+    if (response.ok) {
+
+        const data = await response.json();
+
+        console.log('Successfully fetched data from the PageSpeed Insights API.', data);
+
+        performanceScore = data.lighthouseResult.categories.performance.score * 100;
+        accessibilityScore = data.lighthouseResult.categories.accessibility.score * 100;
+        bestPracticesScore = data.lighthouseResult.categories['best-practices'].score * 100;
+        seoScore = data.lighthouseResult.categories.seo.score * 100;
+
+        const perfRing = document.getElementById('perf-ring') as SlProgressRing
+        if (perfRing) {
+            perfRing.value = performanceScore;
+            perfRing.textContent = performanceScore.toString();
+            perfRing.style.color = performanceScore >= 90 ? 'green' : performanceScore >= 50 ? 'rgb(208, 208, 0)' : 'red';
+            perfRing.style.setProperty("--indicator-color", performanceScore >= 90 ? 'green' : performanceScore >= 50 ? 'yellow' : 'red');
+            if (performanceScore < 90) {
+                perfRing.style.webkitTextStroke = '2px black;'
```

Comment:
Wonder if it's worth putting this into a global const used by all the rings: `ringWidth` or smt I don't know, then if you want to change it you only need to do it in one place

Reply:
There is some other styles aside from what is set here, This is just changing it if the color is not green so that when light-mode is enabled it still is readable....